src/datasets/packaged_modules/imagefolder/imagefolder.py

```diff
@@ -174,14 +194,18 @@ def analyze(files_or_archives, downloaded_files_or_dirs, split):
             else:
                 self.info.features = datasets.Features({"image": datasets.Image()})
 
-            if not self.config.drop_metadata and metadata_files:
-                # Verify that there are no duplicated keys when compared to the existing features ("image", optionally "label")
+            if add_metadata:
+                # Warn if there are duplicated keys in metadata compared to the existing features ("image", optionally "label")
                 duplicated_keys = set(self.info.features) & set(metadata_features)
                 if duplicated_keys:
-                    raise ValueError(
-                        f"Metadata feature keys {list(duplicated_keys)} are already present as the image features"
+                    warnings.warn(
+                        f"Metadata feature keys {list(duplicated_keys)} are already present as the image features, "
+                        f"using image {list(duplicated_keys)} features instead of the metadata ones."
```

Comment:
We prefer warnings from the logger in dataset scripts over `warnings.warn` (+ sentence reformulation to make it easier to understand):
```suggestion
                    logger.warning(
                        f"Ignoring metadata columns {list(duplicated_keys)} as they are already present in the features dictionary."
```

Reply:
@mariosasko I sometimes encounter problems with the datasets logger even if I set logging level manually to info or even debug. so I decided that warning would be better as it is shown independently of logger settings.

Reply:
btw thank you for reformulation!! I'm so bad at this :)

Reply:
Could this issue be related to https://github.com/huggingface/datasets/issues/2832?

PS: You can use the `caplog` fixture in the tests to test the logger's output

Reply:
yes, adding `library_root_logger.addHandler(logging.StreamHandler())` to https://github.com/huggingface/datasets/blob/main/src/datasets/utils/logging.py#L67 fixed the issue!